pandas/tests/reshape/merge/test_merge.py

```diff
@@ -2167,3 +2167,35 @@ def test_merge_datetime_upcast_dtype():
         }
     )
     tm.assert_frame_equal(result, expected)
+
+
+@pytest.mark.parametrize("how", ["left", "right"])
+def test_merge_preserves_row_order(how):
+    # GH 27453
+    population_df = DataFrame(
+        {
```

Comment:
cam you also add an example where merging on the index (both left and right) with how='right' (and 'left') for that matter